File: fixtures/gate-b-session/frontend/src/api/checkout.ts
import { sendJson } from "./client";

export interface CheckoutRequest {
  email: string;
  amount_cents: number;
}

export interface CheckoutResponse {
  order_id: string;
  status: string;
}

export function checkout(request: CheckoutRequest): Promise<CheckoutResponse> {
  return sendJson<CheckoutResponse>("POST", "/orders", request);
}

File: fixtures/gate-b-session/backend/app/routers/checkout.py
from fastapi import APIRouter, status

from ..schemas import CheckoutRequest, CheckoutResponse
from ..store import create_order

router = APIRouter(prefix="/orders", tags=["checkout"])


@router.post("", response_model=CheckoutResponse, status_code=status.HTTP_201_CREATED)
def checkout(payload: CheckoutRequest) -> CheckoutResponse:
    order = create_order(email=payload.email, amount_cents=payload.amount_cents)
    return CheckoutResponse(order_id=order.id, status=order.status)
